### PR TITLE
use explicit build toolset and compiler path in Boost easyblock

### PR DESCRIPTION
**Pull request by Flamefire**
(created using `eb --new-pr`)

Fixes https://github.com/easybuilders/easybuild/issues/695

### CODE REVIEW

**Comment by zao** [comment]
Test report by @zao
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Boost-1.75.0-GCC-10.2.0.eb_ 
 * **SUCCESS** _Boost-1.75.0-iccifort-2020.4.304.eb_ 
 * **SUCCESS** _Boost-1.76.0-iccifort-2020.4.304.eb_ 

Build succeeded for 3 out of 3 (3 easyconfigs in total)
eb-u2004 - Linux Ubuntu 20.04, x86_64, Intel(R) Core(TM) i7-8700K CPU @ 3.70GHz, Python 3.8.5
See https://gist.github.com/a79526fdb18a62eef19edc70f32a7f56 for a full test report.

**Comment by Flamefire** [comment]
Test report by @Flamefire
#### Overview of tested easyconfigs (in order)
 * **SUCCESS** _Boost-1.70.0-gompi-2019a.eb_ 
 * **SUCCESS** _Boost-1.67.0-foss-2018b.eb_ 
 * **SUCCESS** _Boost-1.68.0-intel-2018b-Python-3.6.6.eb_ 
 * **SUCCESS** _Boost-1.71.0-gompi-2019b.eb_ 
 * **SUCCESS** _Boost-1.71.0-iimpi-2019b.eb_ 
 * **SUCCESS** _Boost-1.72.0-gompi-2020a.eb_ 
 * **SUCCESS** _Boost-1.74.0-GCC-10.2.0.eb_ 

Build succeeded for 7 out of 7 (7 easyconfigs in total)
taurusa11 - Linux centos linux 7.7.1908, x86_64, Intel(R) Xeon(R) CPU E5-2603 v4 @ 1.70GHz (broadwell), Python 2.7.5
See https://gist.github.com/94b0d569acff8e34938da92adb4da51d for a full test report.

**Comment by Flamefire** [comment]
@zao @boegel Ping. Do you need any more reports or anything?
This should go into the next release because some are blocked by this.

**Review by akesandgren — APPROVED** [review]
found one more thing, ignore this review

File: easybuild/easyblocks/b/boost.py [hunk 1]
```diff
@@ -139,20 +142,33 @@ def configure_step(self):
...
-            txt = ''
             # Check if using a Cray toolchain and configure MPI accordingly
             if self.toolchain.toolchain_family() == toolchain.CRAYPE:
                 if self.toolchain.PRGENV_MODULE_NAME_SUFFIX == 'gnu':
                     craympichdir = os.getenv('CRAY_MPICH2_DIR')
                     craygccversion = os.getenv('GCC_VERSION')
-                    txt = '\n'.join([
+                    user_config.extend([
                         'local CRAY_MPICH2_DIR =  %s ;' % craympichdir,
                         'using gcc ',
```
Comment (akesandgren) [inline easybuild/easyblocks/b/boost.py:173]: Doesn't this conflict with the toolset setting above? i.e. the user_config.append...

Comment (Flamefire) [reply easybuild/easyblocks/b/boost.py:178]: I'm not sure, maybe the last one wins... But added code to remove it and also check that it is actually used

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @Flamefire!

**Pull request merged**